src/DotLiquid/Tags/Extends.cs

```diff
@@ -103,9 +103,16 @@ public override void Render(Context context, TextWriter result)
         {
             // Get the template or template content and then either copy it (since it will be modified) or parse it
             IFileSystem fileSystem = context.Registers["file_system"] as IFileSystem ?? Template.FileSystem;
-            object file = fileSystem.ReadTemplateFile(context, _templateName);
-            Template template = file as Template;
-            template = template ?? Template.Parse(file == null ? null : file.ToString());
+            Template template = null;
+            if (fileSystem is ITemplateFileSystem)

```

Comment:
It is better to avoid the double cast with:

    var tfs = fileSystem as ITemplateFileSystem;
    if (tfs != null)
    {
        template = tfs.ReadTemplateInstance(context, _templateName);
    }

Or shorter with C#7:

    if (fileSystem is ITemplateFileSystem tfs)
    {
        template = tfs.ReadTemplateInstance(context, _templateName);
    }